From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Wed, 19 Sep 2012 10:13:00 -0400
Subject: include EPICS_MODIFICATION in soname

ABI compatibility between (really any) release is not
guaranteed or checked.

We will have to check any patch releases though.
---
 configure/CONFIG.gnuCommon | 4 ++--
 configure/CONFIG_BASE      | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure/CONFIG.gnuCommon b/configure/CONFIG.gnuCommon
index f054802..aed039b 100644
--- a/configure/CONFIG.gnuCommon
+++ b/configure/CONFIG.gnuCommon
@@ -48,8 +48,8 @@ STATIC_LDFLAGS_YES = -static
 STATIC_LDFLAGS_NO =
 
 SHRLIB_CFLAGS = -fPIC
-SHRLIB_LDFLAGS =  -shared -fPIC
-LOADABLE_SHRLIB_LDFLAGS = -shared -fPIC
+SHRLIB_LDFLAGS =  -shared -fPIC -Wl,-soname,$@
+LOADABLE_SHRLIB_LDFLAGS = -shared -fPIC -Wl,-soname,$@
 
 GNU_LDLIBS_YES = -lgcc
 
diff --git a/configure/CONFIG_BASE b/configure/CONFIG_BASE
index dd55953..1d09c4a 100644
--- a/configure/CONFIG_BASE
+++ b/configure/CONFIG_BASE
@@ -35,8 +35,8 @@ EPICS_BASE_HOST_LIBS += ca Com
 # Version number for base shared libraries (and win32 products)
 
 ifdef BASE_TOP
-SHRLIB_VERSION = $(EPICS_VERSION).$(EPICS_REVISION)
-PROD_VERSION = $(EPICS_VERSION).$(EPICS_REVISION)
+SHRLIB_VERSION = $(EPICS_VERSION).$(EPICS_REVISION).$(EPICS_MODIFICATION)
+PROD_VERSION = $(EPICS_VERSION).$(EPICS_REVISION).$(EPICS_MODIFICATION)
 endif # BASE_TOP
 
 #---------------------------------------------------------------
